src/base/InstanceId.h

```diff
@@ -49,6 +49,7 @@ class InstanceId
     typedef ValueType Value; ///< id storage type
 
     InstanceId() {change();}
+    InstanceId(const InstanceId &); ///< no copying; IDs are unique
```

Comment:
How about marking this explicitly?
No difference in practice, but more clarity on intent

```suggestion
    // work around gcc-14 bug
    InstanceId(const InstanceId &) = delete; ///< no copying; IDs are unique
```